lib/api/api.dart

```diff
@@ -294,17 +294,18 @@ class ApiSingleton {
       }
       if (report.version_number != null) {
         body.addAll({'version_number': report.version_number});
+        if (report.version_number == 0){
+          body.addAll({'phone_upload_time': DateTime.now().toUtc().toIso8601String()});
+        } else if (report.phone_upload_time != null && report.phone_upload_time!.isNotEmpty){
+          body.addAll({'phone_upload_time': report.phone_upload_time});
+        }
```

Comment:
`phone_upload_time` is required in the POST, i'm afraid if version_number is null will never set the field.

I would go to an independent if statement checking that version_number is set and is 0, then set phone_upload_time to now, keep same logic than before

Reply:
I've modified it. But there could be a case where version number is not 0 and phone_upload_time is empty, so it would be passed in the POST and the api call would fail?

Reply:
You're right... But that was the current flow since today... We'll need to improve this in the future.

Reply:
okay, so then we leave it like it's now for the current PR?

Reply:
Yep